### ISSUE TITLE
b:fetchBeanInfos ignores FacesMessages added to context manually

### ISSUE THREAD

**Issue by MitchTalmadge** [TheCoder4eu/BootsFaces-OSP#444]
This is probably intended behavior, but I wanted to ask about it anyway. I feel bad making so many bug reports, I'm curious if there's somewhere I can just chat rather than submitting a new report every time I have a question.

Anyway, my goal is that I am trying to allow validation to be performed within a `b:modal`, and then close the modal once all validations are successful. The `oncomplete` javascript for my `b:commandButton` looks like this: 

`oncomplete="if(!validationFailed) $('.newUserGroupModal').modal('hide')"`

This works great for the bean validations, such as the required length, regex, etc., but when I add a `FacesMessage` from the method that gets called in the action of the `b:commandButton`, the `b:fetchBeanInfos` does not pick up on the message and returns false for `validationFailed`. In my case, I am trying to tell the user that the name they typed in is not valid because something else already has that name; something I cannot (or do not know how to) accomplish using only bean validations. Unfortunately, they never see this message, because the modal is hidden.

Is there a better way to do what I am trying to do? Is it even possible to allow `b:fetchBeanInfos` to pick up on `FacesMessage`s?

Thanks!

**Comment by MitchTalmadge** [comment]
I managed to solve it by using a Validator. Can't believe I never tried that in the first place. :P

**Comment by stephanrauh** [comment]
Actually, I developed the feature using `FacesContect...addMessage()`. So I'm pretty sure it works - see the demo at http://www3.bootsfaces.net/Showcase/forms/FetchBeanInfos.jsf. The second demo allows you to see the values by clicking the command button. The validation result is shown in an alert box.

However, I'm curious what the difference between our implementations is. 

I'd like to close the ticket because the feature you request is already there. We can continue the discussion on our chat https://gitter.im/TheCoder4eu/BootsFaces-OSP. I've added you to the chatroom, so I guess you'll receive an invitation soon.
